homeassistant/components/homekit_controller/event.py

```diff
@@ -0,0 +1,156 @@
+        self._characteristic = service.characteristics_by_type[
+            CharacteristicsTypes.INPUT_EVENT
+        ]
+
+        self.entity_description = entity_description
+
+        # An INPUT_EVENT may support single_press, long_press and double_press. All are optional. So we have to
+        # clamp InputEventValues for this exact device
+        self._attr_event_types = [
+            INPUT_EVENT_VALUES[v]
+            for v in clamp_enum_to_char(InputEventValues, self._characteristic)
+        ]
+
+    def get_characteristic_types(self) -> list[str]:
+        """Define the homekit characteristics the entity cares about."""
+        return [CharacteristicsTypes.INPUT_EVENT]
+
+    async def async_added_to_hass(self) -> None:
+        """Entity added to hass."""
+        await super().async_added_to_hass()
+
+        self.async_on_remove(
+            self._accessory.async_subscribe(
+                [(self._aid, self._characteristic.iid)],
+                self._handle_event,
+            )
+        )
+
+    @callback
+    def _handle_event(self):
+        self._trigger_event(INPUT_EVENT_VALUES[self._characteristic.value])
+        self.async_write_ha_state()
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Homekit event."""
+    hkid: str = config_entry.data["AccessoryPairingID"]
+    conn: HKDevice = hass.data[KNOWN_DEVICES][hkid]
+
+    @callback
+    def async_add_characteristic(char: Characteristic) -> bool:
+        service = char.service
+        entities = []
+
+        if service.type == ServicesTypes.DOORBELL:
+            entities.append(
+                HomeKitEventEntity(
+                    conn,
+                    service,
+                    EventEntityDescription(
+                        key=f"{service.accessory.aid}_{service.iid}",
+                        device_class=EventDeviceClass.DOORBELL,
+                        translation_key="doorbell",
+                        name="Doorbell",
+                    ),
+                )
+            )
+
+        elif service.type == ServicesTypes.SERVICE_LABEL:
+            switches = list(
+                service.accessory.services.filter(
+                    service_type=ServicesTypes.STATELESS_PROGRAMMABLE_SWITCH,
+                    child_service=service,
+                    order_by=[CharacteristicsTypes.SERVICE_LABEL_INDEX],
+                )
+            )
+
+            for idx, switch in enumerate(switches):
+                entities.append(
+                    HomeKitEventEntity(
+                        conn,
+                        switch,
+                        EventEntityDescription(
+                            key=f"{service.accessory.aid}_{service.iid}",
+                            device_class=EventDeviceClass.BUTTON,
+                            translation_key="button",
+                            name=f"Button {idx}",
```

Comment:
This will not work? The name is hardcoded but a translation key is provided. The translation key wins in this case (and could also be omitted to use the device class one). That said, the `{idx}` is thus not used.